src/printer/OITimeToolOutputBuilder.cc

```diff
@@ -59,16 +61,43 @@ std::string OITimeToolOutputBuilder::dump() const {
 }
 
 void OITimeToolOutputBuilder::dumpStatus(std::ostream& ss) const {
-    if (!killReason_.empty()) {
-        ss << killReason_;
+    if (killReason_ != KillReason::NONE) {
+            ss << killReasonComment_;
     }
     else if (killSignal_ > 0) {
         ss << "process exited due to signal " << killSignal_;
     }
+    else if (exitStatus_ > 0) {
+        ss << "runtime error " << exitStatus_;
+    }
     else {
         ss << "ok";
     }
 }
 
+int OITimeToolOutputBuilder::encodeStatusCode() const {
+    static const int CODE_SIG_BASE = 0;
+    static const int CODE_RE_BASE = 200;
+
+    if (killReason_ == KillReason::NONE) {
+        // NOTE: order of ifs is important, as nonzero killSignal also sets exitStatus_
```

Comment:
How about moving code from this if to case KillReason::None below?

Reply:
No. It'd make the switch bulky and asymmetric.

Reply:
Oh, sorry, I haven't noticed if in that else.